src/ott/geometry/costs.py

```diff
@@ -528,70 +531,86 @@ def covariance_fixpoint_iter(
       covs: jnp.ndarray,
       weights: jnp.ndarray,
       tolerance: float = 1e-4,
-      **kwargs: Any
+      sqrtm_kw: Dict[Any, Any] = None,
+      **kwargs
   ) -> jnp.ndarray:
     """Iterate fix-point updates to compute barycenter of Gaussians.
 
     Args:
       covs: [batch, d^2] covariance matrices
       weights: simplicial weights (non-negative, sum to 1)
-      tolerance: tolerance of the overall fixed-point procedure
-      kwargs: keyword arguments for :func:`ott.math.matrix_square_root.sqrtm`.
+      tolerance: tolerance of the fixed-point procedure. That tolerance is
+        applied to the Frobenius norm (normalized by total size)
+        of two successive iterations of the algorithm
+      sqrtm_kw: keyword arguments for :func:`ott.math.matrix_square_root.sqrtm`
+      kwargs: keyword arguments for the outer fixed-point iteration
 
     Returns:
-      Weighted Bures average of the covariance matrices.
+      List containing Weighted Bures average of the covariance matrices, and
+      vector of (normalized) 2-norms of successive differences between iterates,
+      to monitor convergence.
     """
+    sqrtm_kw = {} if sqrtm_kw is None else sqrtm_kw
+    # Pop values or set defaults for fixed-point loop.
+    min_iterations = kwargs.pop("min_iterations", 1)
```

Comment:
Why not just do `kwargs.setdefault("min_iterations", 1)`, etc.?
This way, you can pass `**kwargs` directly to fixed-point iter.

Reply:
Good point! 

I need `max_iterations` and `inner_iterations` to initialize `errors`.
And since those are all positional arguments, maybe it's a bit cuter to expose everyone.